DataScience/ProbabilisticParser/parser.py

```diff
@@ -50,12 +51,11 @@ def parse(raw_string):
     :return: a list of tokens and labels
     :rtype: list
     """
-
-    tokens = t.tokenize(raw_string)
+    tokens = tok.tokenize(raw_string)
```

Comment:
Ah, I see, I mean this kind of logic should normally be covered in methods themselves (and other methods should also return an empty list if the input is an empty list, polymorphism and such). Returning empty list as a features of an empty list is totally normal.